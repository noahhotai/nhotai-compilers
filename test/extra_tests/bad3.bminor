// Returning wrong type in func
main: function integer () = {
    x: integer = 3;
    if (1) {
        for (;;) {
            {
                return 3;
            }
            return 6;
        }
        return "this should be an error";
    }
}
