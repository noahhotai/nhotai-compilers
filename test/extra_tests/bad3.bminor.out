Program parsed successfully.
Program resolved successfully.
if condition must be booleantype error: cannot return a string"this should be an error" in a function (main) that returns integerProgram failed to typecheck.
